### ISSUE TITLE
Report branch coverage (basic blocks executed)

### ISSUE THREAD

**Issue by cobbxia** [SimonKagstrom/kcov#27]
I found the results not analyze branch and class coverage. DO kcov  only supports line coverage ,do we has a plan to support?
I think if we can get the coverage information about lines,we can calculate the branch.The source code must be understood,so there must be a interpreter for  each singe language.

**Comment by SimonKagstrom** [comment]
Closing for now.

**Comment by SimonKagstrom** [comment]
Reopen to track disassembling basic blocks.

**Comment by blackmagic02881** [comment]
wonder if there is any plan to implement this feature? thanks.

**Comment by SimonKagstrom** [comment]
Not really, although the basic block handling is actually implemented now. Presentation-wise I don't know how to handle this though.

**Comment by blackmagic02881** [comment]
that is already a good news. 
we use sonarcube to do the presentation. could you consider to add a support for that first? 
wonder if you could point me to the code where basic block handling was added? 
thanks

**Comment by SimonKagstrom** [comment]
Well, it should be generic enough to allow output everywhere, so sonarcube shoud be no different than other writers. Nice to hear someone using it though!

The interface to the basic block handling is in src/include/disassembler.hh, but currently unused outside of the llvm engine.

**Comment by blackmagic02881** [comment]
Thanks. Will look into it.
